provider/ansible.py

```diff
@@ -163,19 +163,24 @@ def distro_install():
             else:
                 LOG_JOB.error("Ansible repo was required, but failed to be added.")
                 return False
-        install_status = utils_package.package_install('ansible')
+        install_status = utils_package.package_install(package)
         if not install_status:
-            LOG_JOB.error(f"Failed to install ansbile.")
+            LOG_JOB.error(f"Failed to install {package}.")
         yum_backend.remove_repo(params["ansible_repo"])
         return install_status
 
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
-    try:
-        path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
-        ansible_install_policy = params['ansible_install_policy']
-        return policy_map[ansible_install_policy]()
-    else:
-        return True
+    # Dictionary contains command and package providing the command
+    cmd_checklist = {'ansible-playbook': 'ansible', 'sshpass': 'sshpass'}
```

Comment:
`sshpass` is not enough,  what we need to check is the `package_list`:
https://github.com/autotest/tp-qemu/blob/aa0864ad45d5116139facbaaf43e6726a1e79908/provider/ansible.py#L138

With the new approach, function `python_install` need be updated accordingly.

Reply:
@yanan-fu  I prepared this solution only to address the corner case for packages which must be installed from repository (which was the problem with `sshpass`), so IIRC no additional changes for `python_install` should be necessary. Basically it just says: if there is some command missing, try to install it from repository. 

Regarding `package_list`: I see your point, what I intended to address was just checking of commands, that's why I introduced `cmd_checklist` containing command:package key pair. I experimented with the multiple ways how to address it, this one seemed to be just enough. Maybe we can discuss it offline and after put the conclusion here :slightly_smiling_face:

Reply:
@yanan-fu Done, let me summarize the changes:

1. Logic of additional package installation is moved away from `python_install`. This is important for situation ansible is installed via pip *before* the test case is executed at all.
2. `cmd_checklist` is dropped. We need only to check for ansible, so presence of `ansible-playbook`, which was already there.
3. `True` is returned only if ansible and all dependent packages installed correctly.